### DISCUSSION TITLE
solo descarga 10%

### DISCUSSION THREAD

**Discussion by franjichap** [aclap-dev/video-downloadhelper#1044] [Help and Q&A]
### Your question

solo descarga el 10% del principio del video

### Addon & CoApp version and details

llevo años descargando videos de mitele.com,no hace mucho tiempo descargue sin problema con la versión free, con esta misma versión ahora, solo descarga 10 minutos, o un 10% del video, he comprado la licencia por si se tratara de esto, pero no, he intentado descargar de cuatro.com el mismo video y hace lo mismo, he activado vpn y nada cambia, es para lo único que usaba su programa, pero si no funciona,quiero que me devuelvan mi dinero

**Comment by jcfrog** [comment]
please send us an email at licenses@downloadhelper.net

**Comment by Mattmjs** [comment]
No wonder it is not working, I see huge amount of invalid timestamps, invalid DTS messages in ffmpeg.
Example https://www.cuatro.com/todoesmentira/20240722/asesor-partido-democrata-asume-trump-teme-kamala-harris_18_013083541.html

The other website doesn't show anything.

**Reply by GrampaWildWilly** [reply]
[Spanish Video mp4 Log.txt](https://github.com/user-attachments/files/16345417/Spanish.Video.mp4.Log.txt)

Thanks for the URL, @Mattmjs.  I would never have bothered scrounging around for this.  As you can see in that log, it is true that there is a gigantic amount of errors.  Despite that, ffmpeg succeeded in downloading the video.  VLC played it just fine.  I didn't sit & watch it because for one, I don't speak Spanish, & for another, the subject matter didn't interest me anyway.  I have seen other cases where ffmpeg complained to no end but went ahead & created a playable file anyway.  I did use one trick . . . that isn't all that tricky.  Before I did this download, I ran the URL through ffprobe.  I haven't posted that since there is an ffprobe invocation at the start of every ffmpeg invocation.  So you can see the results of ffprobe at the start of the attached log.  You'll notice that the duration quoted for this is N/A.  That is something you see with livestreams.  This is most assuredly not a livestreeam, but I went ahead & treated this as if it were a livestream.  I included the `-m3u8_hold_counters` & `-max_reload` parameters for detection of end-of-stream, which I suspect actually didn't have any effect.  I also included `-live_start_index 0` to rewind the video to its start.  Maybe that had an effect.

All of that just makes it more surprising that VDH couldn't download this, seeing as I used the CoApp's ffmpeg to do that download.  At first, the VDH menu was empty.  But when I clicked play on the first video on the page, suddenly something appeared in the VDH menu.  I didn't have to hit pause because hitting play did not launch playback of the video.  It seemed to cue things up for playback without actually launching playback.  However, VDH downloaded only 15 seconds of content from this.  It didn't throw any errors, just didn't get the whole video.  I don't know why.  There was a perfectly usable HLS master manifest, which I used to do the download you can see in the attached log.  Perhaps my treating it like a livestream, in particular rewinding it to the start, made a difference.

Other web site . . .  Again.  I'm not scrounging around.

**Reply by GrampaWildWilly** [reply]
When I see error messages like these saying the timestamps are invalid, I conclude that the web site encoded the video incorrectly.  They're using some tool, some software, that simply doesn't work according to standards.  The video is damaged & it's probably a testament to the brilliance of the ffmpeg coders that ffmpeg went ahead & downloaded this anyway.

You'll also note a large chunk of the log is occupied trying to work around a bunch of errors that say:

HTTP error 403 Forbidden token format

I have no idea what exactly is going on with that.  But after making some mighty effort, ffmpeg just continued with the rest of the download.  That is very strange.  Those errors occur in the middle of the download.  They aren't at the start, where I would have expected the error to simply abort the download before it begins.  They also aren't at the end, where I would have expected the error to terminate the download prematurely.  So, I think that's very weird.

None of which changes the fact that ffmpeg downloaded this video when I invoked it myself.  So why doesn't VDH download this?  It invokes the same ffmpeg I did.  VDH should have worked here.

**Reply by Mattmjs** [reply]
Maybe your extra parameters help, since when I used ffmpeg the download stopped around 3:30
As an alternative I changed it from **.m3u8** to **.mpd**, there were no errors and downloaded all the way to the end.
The negative was it was throttled to a slow speed. VDH would need all the parameters you used to make it work.

**Reply by JuanmaGonzalez** [reply]
Me sucede lo mismo que a ti @franjichap, lo que antes podía descargar de MiTele, ahora no hay forma de poder descargar más de un 10%...

Y no creo que tengamos alternativa!

**Reply by GrampaWildWilly** [reply]
Of course there's an alternative.  I described it in my post above.  Your choice is simple.  Sit around for some untold length of time waiting for @paulrouget Paul to fix this in VDH.  Or don't use VDH; use ffmpeg instead to get the content now.  This latter alternative would require you to be a student & learn something new.

**Reply by JuanmaGonzalez** [reply]
OK, thanks a lot **@GrampaWildWilly**...
Hopefully this problem will be solved soon.

Regards!
